src/Oscoin/Protocol/Sync/RealWorld.hs

```diff
@@ -85,12 +85,18 @@ newDataFetcher httpManager = DataFetcher $ \peer args ->
                 -- Expected type: IO (ProtocolResponse r)
                 -- Actual type: IO (ProtocolResponse 'GetBlockHeaders)
 
-                let rq = mkRequest GetTip mempty peer
-                liftIO $ HTTP.withResponse rq httpManager $ \res -> do
-                    cborBlob <- HTTP.brRead (HTTP.responseBody res)
-                    case CBOR.deserialise . toS $ cborBlob of
-                      API.Ok t    -> pure t
-                      API.Err err -> throwM $ RequestNetworkError GetTip peerAddr (toS err)
+                response <- C.http (mkRequest GetTip mempty peer) httpManager
+                mbBlock  <- runConduit $ HTTP.responseBody response
+                         .| mapOutput (\r ->
+                             case r of
+                                 API.Ok o  ->
+                                     o :: Block c tx (Sealed c s)
+                                 API.Err e -> panic (toS e) -- any better way?
```

Comment:
Can you add a comment here as to why we panic?

Reply:
Yes -- essentially the reason is that if we get an `Err` there , it means the request didn't succeed, and in theory the impure exception would be caught by the `try` inserted inside `catchNetworkError` and reported as a `SyncError`. Alas that entire expression is pure and I cannot call `throwM` 😏

Reply:
91af9d63